### PR TITLE
Integration: Add Stories to Jetpack XML Sitemap

### LINKED ISSUE #3252: Add Stories to Jetpack XML Sitemap
## Feature Description

The default Sitemap in Jetpack apparently only includes posts, pages, and the custom post types added by Jetpack.

Since we want stories to be discoverable, we need to make sure they're part of that list. Luckily, Jetpack provides a filter for this.

## Alternatives Considered

—

## Additional Context

Related: #3243, #3092 

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

* Jetpack's XML sitemap includes stories

## Implementation Brief

* Filter [`jetpack_sitemap_post_types`](https://developer.jetpack.com/hooks/jetpack_sitemap_post_types/) to add the `web-story` post type.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
Add stories to jetpack sitemaps. 

## Relevant Technical Choices

Also added to WP_COM sitemaps as well, because why not. 

## Testing Instructions

---

Fixes #3252

### CODE REVIEW

File: includes/Story_Post_Type.php [hunk 1]
```diff
@@ -556,4 +565,17 @@ public function filter_site_kit_gtag_opt( $gtag_opt ) {
...
 	}
+
+	/**
+	 * Add CPT to Jetpack sitemaps
+	 *
+	 * @param array $post_types Array of post types.
+	 *
+	 * @return array
+	 */
+	function add_to_sitemap( array $post_types ) {
```
Comment (swissspidy) [inline includes/Story_Post_Type.php]: Let's call this `add_to_jetpack_sitemap`

**Review by swissspidy — CHANGES_REQUESTED** [review]
needs some phpcs fixes

also suggested a different method name.

File: includes/Story_Post_Type.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Story_Post_Type.php]: No, it is just just jetpack sitemaps. See https://github.com/google/web-stories-wp/pull/3264/files#diff-99d793dc332da60eaf6b6846bfd3a7f0R152

Comment (swissspidy) [reply includes/Story_Post_Type.php]: I know, but `add_to_sitemap` is just too ambiguous, and `add_to_jetpack_or_dotcom_sitemap` is clearly not ideal either.

Comment (spacedmonkey) [reply includes/Story_Post_Type.php]: Changed

Comment (swissspidy) [reply includes/Story_Post_Type.php]: I don't think `add_to_jetpack_or_dotcom_sitemap` is ideal though.

I'd say `add_to_jetpack_sitemap` + clear description in the docblock is fine.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**